[Gradient Compression] Directly let world_size = group_to_use.size()

wayi1 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#49715 [Gradient Compression] Directly let world_size = group_to_use.size()**
* #49711 [Gradient Compression] Explicitly restrict the scope of torch.cuda.synchronize to the current device
* #49709 [Gradient Compression] Change wait() to value() in some callbacks of PowerSGD communication hook
* #49451 [Gradient Compression] Warm-start of PowerSGD

Address the comment on https://github.com/pytorch/pytorch/pull/49417#discussion_r545388351

Differential Revision: [D25673997](https://our.internmc.facebook.com/intern/diff/D25673997/)

Commit:
[Gradient Compression] Directly let world_size = process_group.size()

Address the comment on https://github.com/pytorch/pytorch/pull/49417#discussion_r545388351

Differential Revision: [D25673997](https://our.internmc.facebook.com/intern/diff/D25673997/)

[ghstack-poisoned]

rohan-varma commented on torch/distributed/algorithms/ddp_comm_hooks/default_hooks.py:
```diff
@@ -18,9 +18,7 @@ def allreduce_hook(
-    world_size = (
-        process_group.size() if process_group is not None else dist.get_world_size()
-    )
+    world_size = process_group.size()
```
On the line above, it looks like `process_group` can be None. Shouldn't this be `world_size = group_to_use.size()`?

SciPioneer replied:
Yes, obviously I copied and pasted the wrong word :(

Commit:
Update on "[Gradient Compression] Directly let world_size = group_to_use.size()"


Address the comment on https://github.com/pytorch/pytorch/pull/49417#discussion_r545388351

Differential Revision: [D25673997](https://our.internmc.facebook.com/intern/diff/D25673997/)

[ghstack-poisoned]

rohan-varma reviewed (approved):
LGTM, although might want to rebase since some CI looks like it failed with merge conflicts.
